server/command_test.go

```diff
@@ -1,1088 +1,916 @@
-			},
-			setupStore: func(s *mockStore.Store) {
-				s.On("GetLinkByChannelID", testutils.GetChannelID()).Return(&storemodels.ChannelLink{
-					MSTeamsChannel: "Valid-MSTeamsChannel",
-				}, nil).Once()
-				s.On("DeleteLinkByChannelID", testutils.GetChannelID()).Return(nil).Times(1)
-				s.On("GetChannelSubscriptionByTeamsChannelID", "Valid-MSTeamsChannel").Return(nil, errors.New("unable to get the subscription")).Once()
-			},
-			setupClient: func(c *mockClient.Client) {},
-		},
-		{
-			description: "Unable to delete the subscription from the DB",
-			args: &model.CommandArgs{
-				UserId:    testutils.GetUserID(),
-				ChannelId: testutils.GetChannelID(),
-			},
-			setupPlugin: func(p *Plugin) {
-				p.configuration.DisableSyncMsg = true
-			},
-			setupAPI: func(api *plugintest.API) {
-				api.On("GetChannel", testutils.GetChannelID()).Return(&model.Channel{
-					Id:   testutils.GetChannelID(),
-					Type: model.ChannelTypeOpen,
-				}, nil).Times(1)
-				api.On("HasPermissionToChannel", testutils.GetUserID(), testutils.GetChannelID(), model.PermissionManageChannelRoles).Return(true).Times(1)
-				api.On("SendEphemeralPost", testutils.GetUserID(), testutils.GetEphemeralPost("bot-user-id", testutils.GetChannelID(), "The MS Teams channel is no longer linked to this Mattermost channel.")).Return(testutils.GetPost(testutils.GetChannelID(), testutils.GetUserID(), time.Now().UnixMicro())).Times(1)
-			},
-			setupStore: func(s *mockStore.Store) {
-				s.On("GetLinkByChannelID", testutils.GetChannelID()).Return(&storemodels.ChannelLink{
-					MSTeamsChannel: "Valid-MSTeamsChannel",
-				}, nil).Once()
-				s.On("DeleteLinkByChannelID", testutils.GetChannelID()).Return(nil).Times(1)
-				s.On("GetChannelSubscriptionByTeamsChannelID", "Valid-MSTeamsChannel").Return(&storemodels.ChannelSubscription{
-					SubscriptionID: "testSubscriptionID",
-				}, nil).Once()
-				s.On("DeleteSubscription", "testSubscriptionID").Return(errors.New("unable to delete the subscription")).Once()
-			},
-			setupClient: func(c *mockClient.Client) {},
-		},
-	} {
-		t.Run(testCase.description, func(t *testing.T) {
-			p := newTestPlugin(t)
-			mockAPI := &plugintest.API{}
-
-			testCase.setupAPI(mockAPI)
-			p.SetAPI(mockAPI)
-			testutils.MockLogs(mockAPI)
-			defer mockAPI.AssertExpectations(t)
-			testCase.setupPlugin(p)
-			testCase.setupStore(p.store.(*mockStore.Store))
-			testCase.setupClient(p.msteamsAppClient.(*mockClient.Client))
-			_, _ = p.executeUnlinkCommand(testCase.args)
+	th := setupTestHelper(t)
+
+	team := th.SetupTeam(t)
+	user1 := th.SetupUser(t, team)
+	user2 := th.SetupUser(t, team)
+	user3 := th.SetupUser(t, team)
+
+	t.Run("invalid channel", func(t *testing.T) {
+		th.p.configuration.DisableSyncMsg = true
+		commandResponse, appErr := th.p.executeUnlinkCommand(&model.CommandArgs{
+			UserId:    user1.Id,
+			ChannelId: model.NewId(),
```

Comment:
I hope this feels familiar: it's basically what we've been doing in the server forever. Setup a server instance (in this case, a plugin instance: the server + database is a singleton for the package), and just test things directly. No mocking required 🎉.

Reply:
This is awesome :tada: